File: loadbalancer.rb
require "socket"

loadbalancer=TCPServer.open("192.168.182.133",2000)

while true
client=loadbalancer.accept

rhost = client.peeraddr.last
puts "#{rhost} connect to loadbalancer"

fromclient=client.gets
puts "message from client : #{fromclient}"

if fromclient=="boy"
sockettoserver1=TCPSocket.open("192.168.234.129",3000)
sockettoserver1.write(fromclient)
sockettoserver1.close

else if fromclient=="girl"
sockettoserver2=TCPSocket.open("192.168.234.130",4000)
sockettoserver2.write(fromclient)
sockettoserver2.close

else
puts"Error"

end
end
end

File: loadbalancer.java
import java.io.*;
import java.net.*;

public class loadbalancer{
	public static void main(String[]args)throws IOException{
	ServerSocket s1=new ServerSocket(12345);
	while(true){
		Socket connectionsocket=s1.accept();
		BufferedReader messagefromclient=new BufferedReader(new InputStreamReader(connectionsocket.getInputStream()));
		String clientsentence=messagefromclient.readLine();
		
		if(clientsentence.contains("boy")){
		Socket server=new Socket("192.168.234.129",6789);
		DataOutputStream outtoserver=new DataOutputStream(server.getOutputStream());
		outtoserver.writeBytes(clientsentence+"\n");
		server.close();
		}
		
		else if(clientsentence.contains("girl")){
		Socket server2=new Socket("192.168.234.130",2346);
		DataOutputStream outtoserver2=new DataOutputStream(server2.getOutputStream());
		outtoserver2.writeBytes(clientsentence+"\n");
		server2.close();
		}
		else{
		System.out.println("Error");
		}
		}
	}
}
